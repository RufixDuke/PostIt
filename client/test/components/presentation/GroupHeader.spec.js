import React from "react";
import { mount } from "enzyme";
import GroupHeader from '../../../src/components/presentation/GroupHeader';

describe("GroupHeader Component", () => {
  let props;
  let mountedGroupHeader;
  const groupHeader = () => {
    if (!mountedGroupHeader) {
      mountedGroupHeader = mount(
        <GroupHeader {...props} />
      );
    }
    return mountedGroupHeader;
  }

  beforeEach(() => {
    props = {
      notifications: ["New message from neene"],
      children: <div />
    };
    mountedGroupHeader = undefined;
  });

  it("always renders a div", () => {
    const divs = groupHeader().find("div");
    expect(divs.length).toBeGreaterThan(0);
  });

  describe("the rendered div", () => {
    it("contains everything else that gets rendered", () => {
      const divs = groupHeader().find("div");
      const wrappingDiv = divs.first();
      expect(wrappingDiv.children()).toEqual(groupHeader().children());
    });
  });

  describe("rendered `GroupHeader`", () => {
    it("receives props", () => {
      expect(Object.keys(groupHeader().props()).length).toBe(2);
    });
  });

  describe("when notifications is defined", () => {
    beforeEach(() => {
      props.notifications = ["New message from neene"];
    });

    it("checks the class of the rendered notification icon", () => {
      expect(groupHeader().find('.on-notification').exists()).toBe(true);
    });
  });

  describe("when notifications is not defined", () => {
    beforeEach(() => {
      props.notifications = "";
    });

    it("checks the class of the rendered notification icon", () => {
      expect(groupHeader().find('.mybell').exists()).toBe(true);
    });
  });
});
