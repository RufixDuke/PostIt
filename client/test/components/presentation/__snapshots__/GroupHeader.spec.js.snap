// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component displays the group header with notifications in the group section 1`] = `
<GroupHeader
  notifications={
    Array [
      "New message from neene",
    ]
  }
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <ReactTooltip
        insecure={true}
        resizeHide={true}
        wrapper="div"
      >
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
        />
      </ReactTooltip>
      <div
        className="col s12 m12"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="col s12"
            >
              <h5
                className="white-text text-darken-4"
              >
                Your Group List
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/group"
                >
                  <a
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-plus-square add-group"
                      data-tip="Create New Group"
                    />
                  </a>
                </Link>
                <Link
                  className="mybell on-notification"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/notification"
                >
                  <a
                    className="mybell on-notification"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-bell-o"
                      data-tip="You have new Notification(s)"
                    />
                  </a>
                </Link>
              </h5>
            </div>
          </div>
          <hr />
          <div
            className="scrbar center"
          >
            <h5>
              You currently dont belong to any group
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</GroupHeader>
`;

exports[`Message Component displays the group header without notifications 1`] = `
<GroupHeader
  notifications={Array []}
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <ReactTooltip
        insecure={true}
        resizeHide={true}
        wrapper="div"
      >
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
        />
      </ReactTooltip>
      <div
        className="col s12 m12"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="col s12"
            >
              <h5
                className="white-text text-darken-4"
              >
                Your Group List
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/group"
                >
                  <a
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-plus-square add-group"
                      data-tip="Create New Group"
                    />
                  </a>
                </Link>
                <Link
                  className="mybell"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/notification"
                >
                  <a
                    className="mybell"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-bell-o"
                      data-tip="No new Notification"
                    />
                  </a>
                </Link>
              </h5>
            </div>
          </div>
          <hr />
          <div
            className="scrbar center"
          >
            <h5>
              You currently dont belong to any group
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</GroupHeader>
`;
